3rdparty/openssl/CMakeLists.txt

```
@@ -129,146 +276,169 @@ ExternalProject_Add(
 # is put under OPENSSL_BUILD/crypto/perlasm and aesni-mb-x86_64.pl (used to generate
 # aesni-mb-x86_64.s) is put under OPENSSL_BUILD/crypto/aes/asm such that the latter can
 # find the former.
+
+set(OPENSSL_ASSEMBLY_SCRIPTS
+    crypto/aes/asm/aesni-mb-x86_64
+    crypto/aes/asm/aesni-sha1-x86_64
+    crypto/aes/asm/aesni-sha256-x86_64
+    crypto/bn/asm/rsaz-avx2
+    crypto/bn/asm/rsaz-x86_64
+    crypto/bn/asm/x86_64-gf2m
+    crypto/bn/asm/x86_64-mont
+    crypto/bn/asm/x86_64-mont5
+    crypto/ec/asm/ecp_nistz256-x86_64
+    crypto/ec/asm/x25519-x86_64
+    crypto/md5/asm/md5-x86_64
+    crypto/modes/asm/aesni-gcm-x86_64
+    crypto/modes/asm/ghash-x86_64
+    crypto/sha/asm/keccak1600-x86_64
+    crypto/sha/asm/sha1-mb-x86_64
+    crypto/sha/asm/sha1-x86_64
+    crypto/sha/asm/sha256-mb-x86_64
+    crypto/sha/asm/sha512-x86_64)
+
+set(OPENSSL_3_ASSEMBLY_SCRIPTS
```

```suggestion
list(
  APPEND OPENSSL_3_ASSEMBLY_SCRIPTS
  ${OPENSSL_ASSEMBLY_SCRIPTS}
  crypto/aes/asm/aes-x86_64 crypto/aes/asm/bsaes-x86_64
  crypto/bn/asm/rsaz-avx512
```

so that you can get rid of the second list in the first loop below